portfolio/src/main/webapp/script.js

```diff
@@ -106,32 +104,105 @@ function createCommentElement(comment) {
 
   return commentElement;
 }
-async function displayComments() {
+document.addEventListener('DOMContentLoaded', () => {
+  document.getElementById('maxNum').addEventListener('input', () => {
+    let parentElement = document.getElementById('comments');
+    while (parentElement.firstChild) {
+      parentElement.removeChild(parentElement.lastChild);
+    }
+    cursorStr = null;
+    displayComments(cursorStr);
+  });
+});
+
+let cursorStr = null;
+async function displayComments(cursorString) {
   maxNumberOfComments = document.getElementById('maxNum').value;
-  const params = new URLSearchParams();
-  params.append('max_num_of_contents', maxNumberOfComments);
-  const response = await fetch(`/data?max_num_of_comments=${maxNumberOfComments}`);
-  const comments = await response.json();
 
-  commentElement = document.getElementById('comments');
-  while (commentElement.lastElementChild) {
-    commentElement.removeChild(commentElement.lastElementChild);
-  }
+  const response = await fetch(`/data?max_num_of_comments=${maxNumberOfComments}&cursor=${cursorStr}`);
+  const commentData = await response.json();
 
   const commentListElement = document.getElementById('comments');
-  comments.forEach((comment) => {
-    commentListElement.appendChild(createCommentElement(comment));
+  cursorStr = commentData['cursorString'];
+
+  let commentList= commentListElement.children;
+
+  for (let i = 0; i < commentList.length; i++) {
+    commentList[i].style.display = "none";
+  }
+
+  if (commentData['commentList'].length < maxNumberOfComments) {
+    document.getElementById('next-comments').style.display = "none";
```

Comment:
leave a comment for future you. if there's fewer comments than we asked for, that signals that there are no more comments.